### PR TITLE
Feature/deep perception

### PR DESCRIPTION
**Pull request by hello-chintan**
This PR ports to ROS 2 the YOLOv5 object detection and face detection nodes in the stretch_deep_perception package. It also reorganizes the directory structure of the package according to ROS 2 convention.

### CODE REVIEW

File: stretch_deep_perception/package.xml [hunk 1]
```diff
@@ -1,91 +1,58 @@
...
-  <!--   <buildtool_depend>catkin</buildtool_depend> -->
-  <!-- Use exec_depend for packages you need at runtime: -->
-  <!--   <exec_depend>message_runtime</exec_depend> -->
-  <!-- Use test_depend for packages you need only for testing: -->
-  <!--   <test_depend>gtest</test_depend> -->
-  <!-- Use doc_depend for packages you need only for building documentation: -->
-  <!--   <doc_depend>doxygen</doc_depend> -->
-  <buildtool_depend>catkin</buildtool_depend>
   <build_depend>actionlib</build_depend>
   <build_depend>actionlib_msgs</build_depend>
```
Comment (hello-binit) [inline stretch_deep_perception/package.xml:11]: rosdep isn't able to resolve this for galactic

File: stretch_deep_perception/package.xml [hunk 2]
```diff
@@ -1,91 +1,58 @@
 <?xml version="1.0"?>
```
Comment (hello-binit) [inline stretch_deep_perception/package.xml:1]: can we coalesce the various <*_depends> into just the <depends> tag.

File: stretch_deep_perception/setup.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+package_name = 'stretch_deep_perception'
+
+setup(
+    name=package_name,
+    version='0.0.0',
+    packages=[package_name],
+    data_files=[
+        ('share/ament_index/resource_index/packages',
+            ['resource/' + package_name]),
+        ('share/' + package_name, ['package.xml']),
```
Comment (hello-binit) [inline stretch_deep_perception/setup.py:12]: Add

```
        ('share/' + package_name + '/launch', glob('launch/*.launch.py')),
        ('share/' + package_name + '/rviz', glob('rviz/*')),
```
```

**Comment by hello-binit** [comment]
@hello-chintan The launch files for detect object and detect faces are missing. Can you open another PR to add these and address the above comments?

File: stretch_deep_perception/package.xml [hunk 4]
```diff
@@ -1,91 +1,58 @@
...
-  <exec_depend>rospy</exec_depend>
+  <exec_depend>rclpy</exec_depend>
   <exec_depend>std_msgs</exec_depend>
   <exec_depend>std_srvs</exec_depend>
-  <exec_depend>tf</exec_depend>
   <exec_depend>tf2</exec_depend>
-  <exec_depend>ros_numpy</exec_depend>
+  <exec_depend>ros2_numpy</exec_depend>
+  <exec_depend>python-pytorch-pip</exec_depend>
+  <exec_depend>python-seaborn</exec_depend>
```
Comment (hello-binit) [inline stretch_deep_perception/package.xml:48]: rosdep is not resolving this

File: stretch_deep_perception/package.xml (hunk 1 above)
Comment (hello-chintan) [reply stretch_deep_perception/package.xml:11]: Removed dep on actionlib in PR #28.

File: stretch_deep_perception/package.xml (hunk 4 above)
Comment (hello-chintan) [reply stretch_deep_perception/package.xml:48]: Removed dep on ros2_numpy for this package in PR #28.

File: stretch_deep_perception/setup.py (hunk 3 above)
Comment (hello-chintan) [reply stretch_deep_perception/setup.py:12]: Added paths in PR #28.

File: stretch_deep_perception/package.xml (hunk 2 above)
Comment (hello-chintan) [reply stretch_deep_perception/package.xml:1]: Updated all tags to <depend> in PR #28.

**Pull request merged**